### PR TITLE
Add dynamic trigger aria-label support to Collapsible

### LINKED ISSUE #555: Collapsible: trigger does not support dynamic aria-label
## Problem

The `Collapsible` trigger element does not expose a prop to set a dynamic `aria-label`. Test suites that query collapsible sections by their `aria-label` text (e.g. via `getByRole('button', { name: '...' })`) cannot target the trigger reliably when the label changes based on runtime state.

## Repro

Use `Collapsible` where the trigger label needs to reflect dynamic content — there is no `triggerLabel` / `aria-label` prop forwarded to the trigger button.

## Expected behaviour

The `Collapsible` component should forward (or expose) an `aria-label` prop on the trigger element so consumers can set an accessible, dynamic label.

## Context

Discovered while migrating `ConversationView.svelte` in the Stardust project. Tool-call detail panels use `<details>` with a dynamic `aria-label` that tests assert against; `Collapsible` cannot replace them without this feature.

### PR DESCRIPTION
**Pull request by stevekinney**
## Why

`Collapsible` could not expose a dynamic accessible name on its internal trigger button, which blocked migration from `<details>` patterns that rely on state-aware `aria-label` assertions. This change also locks in the `open` control behavior expected by consumers replacing `<details open>`.

## What changed

- Added `triggerAriaLabel` to `Collapsible` as a dedicated trigger-level prop.
  - Supports a static string or a state-aware function receiving `{ open, disabled }`.
  - Forwards to the internal trigger button's `aria-label` so the accessible name can update reactively.
- Expanded Collapsible regression coverage:
  - static and dynamic `triggerAriaLabel` behavior,
  - parent-driven `open` prop updates,
  - existing `bind:open` two-way behavior remains covered.
- Updated Collapsible docs and generated schema artifacts to reflect the public contract.

## Notes for reviewers

- This keeps the current open-state model explicit: parents can drive `open` directly, and `bind:open` is optional for two-way synchronization.
- Includes documentation/schema updates so discoverability and generated props tables stay aligned with runtime behavior.

Fixes: #555
Related: #554

---

> [!NOTE]
> **Low Risk**
> Focused disclosure/a11y API addition with tests and documentation; no auth, data, or infrastructure changes.
> 
> **Overview**
> Adds **`triggerAriaLabel`** on `Collapsible` so the internal trigger can expose a static or state-aware accessible name (`{ open, disabled } => string`), mapped to `aria-label` with whitespace-only values omitted.
> 
> **Panel labeling** now uses a dedicated `{base}-label` element for `aria-labelledby` instead of the trigger button id, so a trigger-only `aria-label` override does not change the panel region’s landmark name.
> 
> Docs, JSON schema, and a11y notes are updated to describe **`open`**: without `bind:open`, parent prop updates still drive the panel while trigger clicks stay local unless binding is used. Regression tests cover `triggerAriaLabel`, parent-driven `open`, and the new label wiring.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit cb5aa1636776aa7c63255aa205c4eec4588797a1. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/components/src/components/collapsible/collapsible.svelte [hunk 1]
```diff
@@ -44,6 +45,11 @@
   const baseId = $derived(idBase ?? autoId);
   const headerId = $derived(`${baseId}-header`);
   const panelId = $derived(`${baseId}-panel`);
+  const computedTriggerAriaLabel = $derived.by(() =>
+    typeof triggerAriaLabel === 'function'
+      ? triggerAriaLabel({ open, disabled })
+      : triggerAriaLabel,
+  );
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/collapsible/collapsible.svelte]: Good catch. I normalized `triggerAriaLabel` before rendering so empty or whitespace-only values now omit `aria-label` instead of suppressing the accessible-name fallback. I also added a regression test that verifies whitespace results do not emit the attribute.

File: packages/components/src/components/collapsible/collapsible-open-prop-harness.svelte [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<!--
+  Test-only harness pinning the unbound `open` contract: `open={...}` seeds the
+  initial state, and later parent writes do not drive the panel unless `bind:open`
+  is used. Not a public component.
+-->
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/collapsible/collapsible-open-prop-harness.svelte:5]: Updated this harness header comment to match the actual contract: parent writes can drive visibility via `open={...}`. The wording no longer claims parent writes are ignored without `bind:open`.

File: packages/components/src/components/collapsible/README.md [hunk 3]
```diff
@@ -32,9 +32,10 @@ Controlled — the parent owns the state with `bind:open`:
 
 ### State model
 
-Without `bind:open`, `open` is the initial local value and the component manages subsequent toggles; pass `open={true}` to render initially open. With `bind:open`, the parent owns the state. `ontoggle` fires on every toggle with the next boolean, so one-way observers stay in sync.
+`open` is bindable and can be driven directly by parent updates (`open={...}`), including initial-open use cases like replacing `<details open>`. `bind:open` adds two-way synchronization so trigger clicks flow back into parent state automatically. `ontoggle` fires on every toggle with the next boolean, so one-way observers stay in sync.
 
 The `trigger` prop is either a string or a snippet receiving `{ open, disabled }`, letting the label react to state (for example, swapping "Show" and "Hide").
+Use `triggerAriaLabel` when you need a dedicated accessible name on the internal trigger button (for example, icon-heavy labels or test selectors based on `getByRole(..., { name })`).
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/collapsible/README.md]: Agreed. I expanded the README text to explicitly document the state-aware function form for `triggerAriaLabel` (`({ open, disabled }) => string`) alongside the static string form.

File: packages/components/src/components/collapsible/collapsible.schema.ts [hunk 4]
```diff
@@ -11,14 +11,18 @@ const schema = {
...
       default: false,
     },
     disabled: {
       type: 'boolean',
       description: 'When true, the trigger cannot be toggled.',
       default: false,
     },
+    triggerAriaLabel: {
+      type: 'string',
+      description: 'Accessible name override for the trigger button.',
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/collapsible/collapsible.schema.ts]: Updated the schema-source description to call out that runtime supports the state-aware function form, while JSON Schema models only the string variant. Regenerated artifacts now carry that clarification.

File: packages/components/src/components/collapsible/collapsible.schema.json [hunk 5]
```diff
@@ -8,14 +8,18 @@
...
       "default": false
     },
     "disabled": {
       "type": "boolean",
       "description": "When true, the trigger cannot be toggled.",
       "default": false
     },
+    "triggerAriaLabel": {
+      "type": "string",
+      "description": "Accessible name override for the trigger button."
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/collapsible/collapsible.schema.json]: Updated. The generated schema description now notes that the runtime API also supports a state-aware function even though the schema type remains `string`.

File: packages/components/src/components/collapsible/README.md [hunk 6]
```diff
@@ -32,9 +32,10 @@ Controlled — the parent owns the state with `bind:open`:
 
 ### State model
 
-Without `bind:open`, `open` is the initial local value and the component manages subsequent toggles; pass `open={true}` to render initially open. With `bind:open`, the parent owns the state. `ontoggle` fires on every toggle with the next boolean, so one-way observers stay in sync.
+`open` is bindable and can be driven directly by parent updates (`open={...}`), including initial-open use cases like replacing `<details open>`. `bind:open` adds two-way synchronization so trigger clicks flow back into parent state automatically. `ontoggle` fires on every toggle with the next boolean, so one-way observers stay in sync.
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/collapsible/README.md]: Thanks, agreed. I tightened the state-model docs to avoid presenting unbound `open={...}` as fully controlled: docs now state that parent prop writes can update open, but trigger clicks remain local unless `bind:open` (or parent `ontoggle` mirroring) is used.

File: packages/components/src/components/collapsible/collapsible.svelte [hunk 7]
```diff
@@ -63,6 +74,7 @@
     type="button"
     id={headerId}
     class="cinder-collapsible__trigger"
+    aria-label={computedTriggerAriaLabel}
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/collapsible/collapsible.svelte:78]: Good point. I changed the panel labeling to use a dedicated visible-label id (`{base}-label`) and pointed `aria-labelledby` there, so trigger-only `triggerAriaLabel` overrides no longer rename the panel region.

File: packages/components/src/components/collapsible/collapsible.schema.json [hunk 8]
```diff
@@ -8,14 +8,18 @@
...
       "description": "When true, the trigger cannot be toggled.",
       "default": false
     },
+    "triggerAriaLabel": {
+      "type": "string",
+      "description": "Accessible name override for the trigger button. The runtime prop also\naccepts a state-aware function (`{ open, disabled } => string`), but JSON\nSchema can only model the string variant."
+    },
     "idBase": {
       "type": "string",
       "description": "Base used to derive the trigger and panel ARIA ids. Auto-generated when omitted."
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/collapsible/collapsible.schema.json]: Agreed. I updated the source prop documentation so `idBase` now includes the derived label id, then regenerated artifacts so the schema JSON reflects the full `${base}-header`, `${base}-label`, and `${base}-panel` contract.

File: packages/components/src/components/collapsible/README.md [hunk 9]
```diff
@@ -49,15 +50,16 @@ For multiple coordinated sections where opening one may close others, use [`Acco
...
+| ------------------ | ---------- | -------- | ------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `class`            | `string`   | no       | —       | Additional classes merged onto the root element.                                                                                                                                           |
+| `disabled`         | `boolean`  | no       | `false` | When true, the trigger cannot be toggled.                                                                                                                                                  |
+| `idBase`           | `string`   | no       | —       | Base used to derive the trigger and panel ARIA ids. Auto-generated when omitted.                                                                                                           |
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/collapsible/README.md]: Fixed. The generated README props table now documents that `idBase` derives trigger, label, and panel ids, matching the current ARIA wiring.
